### ISSUE TITLE
Question: Can AHK++ be made to determine the editors cursor postion?

### ISSUE THREAD

**Issue by fade2gray** [mark-wiemer/vscode-autohotkey-plus-plus#5]
Hello,

I posted a similar [url=https://stackoverflow.com/q/64791832/14619965]question[/url] on Stack Overflow but it hasn't garnered much interest.

Essentially, I'm trying to determine what the start position is, when the copy command is used, for a selected block of text, preferably using windows cmd/Powershell to be used as a script variable.

![BlockSelect](https://user-images.githubusercontent.com/1332009/98961419-739e1d80-24fd-11eb-9ebb-b2afa181699d.PNG)

I found this [API Reference](https://code.visualstudio.com/api/references/vscode-api#Position) and wondered if it could be used in some fashion.

Thanks for any help.

**Comment by mark-wiemer** [comment]
Hi, sorry for the late reply. Are you able to provide some sort of pseudocode JSON output of what you're expecting? Do you want me to expose something like `{ row: 2, col: 4 }` when you press `Ctrl + C`?

Looking at your SO question, it looks like you want to copy both the index of the starting position _and_ the selected text in one "Copy" command, is that right? If so, this sounds like a very niche use case, and I can't say I'd spend much time working on it. The purpose of this tool is to help developers write AHK code, and I don't see how that would help. You're more than welcome to help me understand how this is useful!

**Comment by fade2gray** [comment]
Hi, my use case was to produce an AHK script that could simplify the creation of folding block comments.

As it is, VSC requires lines of text within the block comment to be indented for it to fold. The need for indentation can be eliminated with the combined use of the folding region tags `;region  ;endregion`.; the tag combination can be achieved by removing the circumflex anchors from lines 34 & 35 of the `ahk.configuration.json` file.

For the script to function correctly, it would need to ensure the selected block of text starts at column 0 and finishes at the end of a  line and would otherwise issue an alert when launched.

Perhaps this concept could be encoded within the extension?

![FoldingBlockComment](https://user-images.githubusercontent.com/1332009/101901412-ea841000-3ba8-11eb-96d2-ae951943ebd3.PNG)

**Comment by mark-wiemer** [comment]
I saw your commit to `cweijan`'s repo, and I appreciate the contribution! I plan to review the changes he's made and merge them into mine, including the removal of the anchors that you mentioned. Folding block comments are definitely important to me, and I think I can do even better than the implementation you've shown in your screenshot. Instead of:

```ahk
/* ;region
line one
line two
*/ ;endregion
```

How about:

```ahk
/*
line one
line two
*/
```

And the above comment is foldable?

**Comment by FuPeiJiang** [comment]
what extension are you using to make block comments foldable ?
currently I'm using this (toggle line comments):
https://stackoverflow.com/questions/60334430/shortcut-to-toggle-commented-and-uncommented-code-at-the-same-time#60366506
![](https://i.stack.imgur.com/uX1iy.gif)

**Comment by fade2gray** [comment]
@mark-wiemer 

> I saw your commit to cweijan's repo, and I appreciate the contribution!

I made a further commit [here](https://github.com/AutoHotkey-Plus/vscode-autohotkey/pull/64). I edited the RegEx's like so...
```
"start": "^(\\/\\*)?\\s*\\;\\s*region\\b",
"end": "^(\\*\\/)?\\s*\\;\\s*endregion\\b"
```

> How about:
> 
> /*
> line one
> line two
> */
> 
> And the above comment is foldable?

Yes, just like in Notepad++ and similar editors.

This is a [folding 'helper' script](https://github.com/fade2gray/VSCode-AutoHotkey-Easy-Folding/blob/main/README.md) I devised.

**Comment by FuPeiJiang** [comment]
@fade2gray I answered your question, but on stack overflow
https://stackoverflow.com/questions/64791832/found-somevscode-how-can-i-determine-the-cursors-column-position-within-an-edi#65316198

**Comment by fade2gray** [comment]
@FuPeiJiang Yes, thanks for that. I didn't reply there because it says not to leave comments just to say "thanks". Now, with my extremely limited experience with creating extensions, I need to try and implement your code suggestion.

Thanks again. 👍

**Comment by FuPeiJiang** [comment]
I think you already solved it using ahk ? did you ?

https://code.visualstudio.com/api/get-started/your-first-extension
yo code
I choose javascript extension because it is faster than typescript to test
on the left, in debug, click the green button
![image](https://user-images.githubusercontent.com/42662615/102415781-c40a0e80-3fc6-11eb-8cce-41fb4da8439d.png)

![image](https://user-images.githubusercontent.com/42662615/102415469-36c6ba00-3fc6-11eb-98bb-6d890cfbad5c.png)
and then
ctrl+shift+p to run a command, the command name defaults to "Hello World"

idk yet how to edit file using vscode, how to copy to clipboard

**Comment by FuPeiJiang** [comment]
using ahk, if you want to block comment lines without selecting the full line, (without selecting start of line, without selecting end of line)
```
f1::
sleep, 50
mass:=getClip()
send, +{home}
left:=getClip()
send, +{end}
right:=getClip()
finalstr:=";region `r`n" left mass right "`r`n;endregion"
clipboard:=finalstr

send, ^v

arr:=StrSplit(finalstr,"`n", "`r")
loop % arr.Length() - 1 {
    send, {up}
}
send, {home}
loop 8
send,{right}
return
; Reload

getClip()
{
    clipboard:=""
    Send, {ctrl down}x{ctrl up}
    sleep, 50
return clipboard
}
```

**Comment by mark-wiemer** [comment]
Closed by #27
